core/src/test/scala/integration/kafka/api/TransactionsTest.scala

```diff
@@ -352,6 +351,7 @@ class TransactionsTest extends KafkaServerTestHarness {
       }
     } finally {
       consumer.close()
+      producer1.close()
```

Comment:
We can do it this way for now, but one thing that we've done to make tests more robust is to use a factory method that registers the instance in a collection. During tearDown, we can then close everything.